src/abstractitensornetwork.jl

```diff
@@ -603,12 +603,13 @@ function gauge_edge(
   # # TODO: Implement as `only(common_neighbors(tn, src(edge), dst(edge)))`
   # new_vertex = only(neighbors(tn, src(edge)) ∩ neighbors(tn, dst(edge)))
   # return contract(tn, new_vertex => dst(edge))
+  @assert src(edge) ∈ neighbors(tn, dst(edge))
```

Comment:
It is best to avoid `@assert`, it is meant to be used for low level checks of things that should never happen, rather than checking bad inputs, and could get removed by certain optimization levels (https://docs.julialang.org/en/v1/base/base/#Base.@assert). I think the right error to throw is `throw(ArgumentError("Edge not in graph."))` since that is the error to throw for bad arguments.